### ISSUE TITLE
ESM modules

### ISSUE THREAD

**Issue by grosch** [just-jeb/angular-builders#1226]
With Angular 14 and jest-preset-angular 12 it's now possible to get pretty far along the ESM path. Will this be upgraded to give me an option to use that instead of the commonjs versions?

In the jest-preset-angular packages there's an examples/examples-v14 directory where you can see the versions of the files with the -esm path component in them.

**Comment by just-jeb** [comment]
Yes, the Jest builder will support ESM. However, it's not a small task so it will require some time.

**Comment by grosch** [comment]
@just-jeb Any ETA on when this might happen?

**Comment by muuvmuuv** [comment]
Hmm, any help you need here? Wondering if we can get this working easier now since Angular moves more and more to ESM configuration with verbatim modules.

**Comment by muuvmuuv** [comment]
Ok... just found out... using vitest is just... a no brainer :D had to switch just insane (didnt even needed to adjust any test spec). Copied most of the code from AnalogJS and use their manuall installation but it works super smooth. One just need to make sure to install the exact same vite version as `@angular/build` installs. (https://analogjs.org/docs/features/testing/vitest#setup-for-running-tests-for-node)

**Comment by livanov** [comment]
Bumping up the issue :) a support for ESM would be much appreciated :)

**Comment by just-jeb** [comment]
Closing — Angular 21 adopted Vitest natively which solves the ESM story properly. For users wanting ESM today, vitest is the recommended path; for sticking with Jest, the CJS preset still works.
